### ISSUE TITLE
Cannot run the arm64 program cross-compiled by Rust on Surface Pro X (0xc000007b)

### ISSUE THREAD

**Issue by zhmushan** [patriksvensson/blog#1]
@patriksvensson Sorry to disturb you here, but I really can’t make progress (I’ve spent several days on this).

First, I installed rust (x86_64) on the Surface (the arm64 windows can run x86_64 programs now).
Then I installed build tools and windows sdk, see the pictures below.

![image](https://user-images.githubusercontent.com/24505451/107212783-6d5b0700-6a42-11eb-8eb6-55a9d0087303.png)
![image](https://user-images.githubusercontent.com/24505451/107212735-5ae0cd80-6a42-11eb-88ef-0aa8c5b10558.png)

I use the `nightly-x86_64-pc-windows-msvc` toolchain default, and the compilation target is set to `aarch64-pc-windows-msvc`.
```toml
# config.toml
[target.aarch64-pc-windows-msvc]
linker = "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/bin/Hostx64/arm64/link.exe"
rustflags = [
  "-L", "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.18362.0/um/arm64",
  "-L", "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/lib/arm64",
  "-L", "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.18362.0/ucrt/arm64"
]

[target.x86_64-pc-windows-msvc]
linker = "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/link.exe"
rustflags = [
  "-L", "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.18362.0/um/x64",
  "-L", "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/lib/x64",
  "-L", "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.18362.0/ucrt/x64"
]

[target.i686-pc-windows-msvc]
linker = "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x86/link.exe"
rustflags = [
  "-L", "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.18362.0/um/x86",
  "-L", "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/lib/x86",
  "-L", "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.18362.0/ucrt/x86"
]
```

In the end, I believe that my programs (helloworld) are compiled successfully, but the program whose target is `aarch64-pc-windows-msvc` cannot run on my Surface Pro X.

![image](https://user-images.githubusercontent.com/24505451/107213406-52d55d80-6a43-11eb-96a7-b143c2f97c8f.png)

I will be very grateful if you can give me some hints.

**Comment by patriksvensson** [comment]
@zhmushan Have you installed the `Microsoft Visual C++ Redistributable for Visual Studio 2019 for ARM64` on your Surface Pro X? https://aka.ms/vs/16/release/VC_redist.arm64.exe

**Comment by zhmushan** [comment]
![image](https://user-images.githubusercontent.com/24505451/107216152-4c48e500-6a47-11eb-8f03-729d86cf311a.png)
@patriksvensson I have installed.

**Comment by zhmushan** [comment]
I have also tried to uninstall redist.x86 and only keep redist.arm64, but this does not work, it will only cause some x86 applications to fail to run.

**Comment by patriksvensson** [comment]
@zhmushan Have you tried running stable instead of nightly?

**Comment by zhmushan** [comment]
I tried it immediately, unfortunately, it still doesn't work

**Comment by patriksvensson** [comment]
@zhmushan Ok, I don't really know what's going on. I think your best bet is to install windbg on the Surface X and see what is happening when it starts.

**Comment by patriksvensson** [comment]
@zhmushan Btw, 0x000007b indicates that the app might try to load a dll with the wrong architecture. Are you sure you've copied the correct executable to the ARM64 device?

**Comment by zhmushan** [comment]
> @zhmushan Btw, 0x000007b indicates that the app might try to load a dll with the wrong architecture. Are you sure you've copied the correct executable to the ARM64 device?

Of course, I am running x86 rustc on an arm64 device, through cross-compilation to generate an arm64 executable file. That is, all operations are done on the same device.

> @zhmushan Ok, I don't really know what's going on. I think your best bet is to install windbg on the Surface X and see what is happening when it starts.

Thank you very much for your help, I will try to use windgb to troubleshoot the error.
